### PR TITLE
Field dim vdim mapping

### PR DESCRIPTION
**Pull request by lang-m**
@samjrholt I would like to get your opinion on this before adding it to all the class creations.

### CODE REVIEW

**Comment by lang-m** [comment]
Maybe `vdim_mapping` would be a better name?

File: discretisedfield/field.py [hunk 1]
```diff
@@ -524,6 +528,35 @@ def valid(self, valid):
 
         self._valid = _as_array(valid, self.mesh, nvdim=1, dtype=bool)
 
+    @property
+    def dim_mapping(self):
+        """Map vdims to dims."""
+        return self._dim_mapping
+
+    @dim_mapping.setter
+    def dim_mapping(self, mapping):
```
[the remark this answers is not in the document]
Comment (samjrholt) [reply discretisedfield/field.py:537]: Actually this may complicate things for the `x0`, `v0` case.

**Comment by samjrholt** [comment]
I prefer `vdim_mapping` I think.

File: discretisedfield/field.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lang-m) [reply discretisedfield/field.py:537]: I would rather just rely on the order as this is very clear and hopefully comes without surprises. And `x0` + `v0` would be difficult.

**Comment by lang-m** [comment]
Do we want to save `vdim_mapping` in the xarray.DataArray or loose it?

**Comment by samjrholt** [comment]
I think keep it

File: discretisedfield/tests/test_field.py [hunk 2]
```diff
@@ -635,6 +635,67 @@ def test_norm_zero_field():
     assert np.all(f.norm.array == 0)
 
 
+@pytest.mark.parametrize(
+    "p1, p2, n, nvdim, dim_mapping, dim_mapping_check",
```
Comment (samjrholt) [inline discretisedfield/tests/test_field.py:639]: Looks all good! Very very minor but should this be renames `vdim_mapping` now?

**Review by samjrholt — APPROVED** [review]

**Pull request merged**